### PR TITLE
Ranges Algorithms Modernization - Return

### PR DESCRIPTION
**Pull request by mitaclaw**
In C++20, algorithms from the standard library evolved to feature new niceties and additional compile-time checks. A few of these updated algorithm functions (found in the `std::ranges` namespace) have new return values, and thus code beyond the expression being modernized must also be updated. I have isolated these changes so that they may be recognized and more easily understood.

### CODE REVIEW

**Comment by mitaclaw** [comment]
This PR will remain a draft until I am confident there are no mistakes in my changes.

Special shout-out to the PR that inspired me to attempt this modernization: https://github.com/dolphin-emu/dolphin/pull/12992

**Comment by BhaaLseN** [comment]
Those `std::erase` changes seem a lot more verbose, is that really worth it?

**Comment by mitaclaw** [comment]
It's definitely subjective. The impression that the standard library's algorithm design gives me is that the returned subrange is the intended way to obtain the first and last iterator for erasing. It was a little unfortunate that this required a new variable in three places, but I preferred the newer design for being straightforward. I think it's worth it aesthetically, but if it's an eyesore to everyone else I can revert to obtaining the end iterator from the original container.

Also, I failed to notice this before (I guess I was a little sleep deprived?), but I actually fixed UB in `OpenModeToAndroid` with my changes. Previously, any openmode string that didn't have a 'b' character in it would erase the end iterator, which is [no good](https://en.cppreference.com/w/cpp/string/basic_string/erase).

**Comment by mitaclaw** [comment]
Rebase with no changes; the buildbots only failed the first time because they timed out.

**Review by sepalani — COMMENTED** [review]
If you prefer, you can also use the PairLike operator for the unique results instead of calling `begin()` and `end()`:
```C++
  const auto [begin, end] = std::ranges::unique(result);
  result.erase(begin, end);
```

**Comment by BhaaLseN** [comment]
I don't have any strong feelings either way, it's just that most of the time, _longer_ code isn't always the _better_ code; idiomatic or not.
That being said, don't let me block you from this.

However, I was curious and tried sepalani's suggestion with `std::erase`. It works for the string, but I couldn't find a way to make the two vectors work with `std::ranges::unique` while keeping it fairly concise.

**Comment by mitaclaw** [comment]
`std::erase` for containers is a shorthand for the remove-erase idiom, so it would only work for that example. Good suggestion, by the way!

**Comment by mitaclaw** [comment]
I really hoped I could manually test every algorithm I was changing, at least for the non-trivial changes, but even among this small sampling many of them are for edge-cases that I don't know how to reproduce. I have reviewed my understanding of the algorithms, toyed around in Compiler Explorer with similar usages, and closely reread the diffs. I think everything is right.

File: Source/Core/DolphinQt/Settings.cpp [hunk 1]
```diff
@@ -335,11 +335,11 @@ void Settings::RemovePath(const QString& qpath)
   std::string path = qpath.toStdString();
   std::vector<std::string> paths = Config::GetIsoPaths();
 
-  auto new_end = std::remove(paths.begin(), paths.end(), path);
-  if (new_end == paths.end())
+  const auto remove_result = std::ranges::remove(paths, path);
```
Comment (iwubcode) [inline Source/Core/DolphinQt/Settings.cpp:338]:
```
if (std::erase(paths, path) == 0)
  return;
```

?

Comment (mitaclaw) [reply Source/Core/DolphinQt/Settings.cpp:338]: Reviewing this code again, the early return here is awfully paranoid. Not my concern, though.

**Comment by mitaclaw** [comment]
Rebase with no changes; ebf7cebc32ac9af28dc2383192b7b14f7aea134f modified algorithms directly next to uses of `std::unique`.

**Review by dreamsyntax — APPROVED** [review]
LGTM. The changes here are more intuitive imo.

**Review by sepalani — APPROVED** [review]
Code LGTM, and tested on Windows.
